fontir/src/glyph.rs

```diff
@@ -288,56 +368,36 @@ impl Work<Context, WorkError> for FinalizeStaticMetadataWork {
         // 1) need to push their paths to a new glyph that is a component
         // 2) collapse such glyphs into a simple (contour-only) glyph
         // fontmake (Python) prefers option 2.
-        for (glyph_to_fix, has_consistent_2x2_transforms) in current_metadata
-            .glyph_order
-            .iter()
-            .map(|gn| context.get_glyph_ir(gn))
-            .map(|glyph| {
-                let consistent_transforms = has_consistent_2x2_transforms(&glyph);
-                (glyph, consistent_transforms)
-            })
-            .filter(|(glyph, has_consistent_2x2_transforms)| {
-                !has_consistent_2x2_transforms || has_components_and_contours(glyph)
-            })
-        {
-            if !has_consistent_2x2_transforms || context.flags.contains(Flags::MATCH_LEGACY) {
-                if !has_consistent_2x2_transforms {
+        for glyph_name in current_metadata.glyph_order.iter() {
+            let glyph = context.get_glyph_ir(glyph_name);
+            let inconsistent_components = !has_consistent_2x2_transforms(&glyph);
+            if inconsistent_components || has_components_and_contours(&glyph) {
+                if inconsistent_components {
                     debug!(
                         "Coalescing'{0}' into a simple glyph because component 2x2s vary across the designspace",
-                        glyph_to_fix.name
+                        glyph.name
                     );
-                } else {
+                    convert_components_to_contours(context, &glyph)?;
+                } else if context.flags.contains(Flags::PREFER_SIMPLE_GLYPHS) {
                     debug!(
-                        "Coalescing'{0}' into a simple glyph because it has contours and components and that's how fontmake handles it",
-                        glyph_to_fix.name
+                        "Coalescing'{0}' into a simple glyph because it has contours and components and prefer simple glyphs is set",
```

Comment:
```suggestion
                        "Coalescing '{0}' into a simple glyph because it has contours and components and prefer simple glyphs is set",
```

Reply:
also missing a space in the previous `debug!("Coalescing'{0}'...`